src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -444,11 +444,16 @@ else if (kruizeObject.getMode().equalsIgnoreCase(AnalyzerConstants.AUTO) || krui
                         experimentName, interval_start_time, interval_end_time));
                 kruizeObject.setValidation_data(new ValidationOutputData(false, e.getMessage(), HttpServletResponse.SC_INTERNAL_SERVER_ERROR));
             }
-        } catch (Exception | FetchMetricsError | InvalidModelException | InvalidTermException e) {
+        } catch (Exception | FetchMetricsError e) {
```

Comment:
The superclass `Exception` should come at the last else it will not handle your exceptions correctly.